--- FreeWRL-1.13/Makefile.PL.orig	2005-04-25 15:47:19.000000000 +0200
+++ FreeWRL-1.13/Makefile.PL	2005-04-28 22:16:18.330867064 +0200
@@ -412,8 +384,6 @@
 			push(@postamble, join(
 				"\n",
 				qq{# make the libFreeWRLFunc shared library active\n},
-				qq{\tcp blib/arch/auto/VRML/VRMLFunc/VRMLFunc.so \$(DESTDIR)$LIB_DIR/libFreeWRLFunc.so\n},
-				qq{\t/sbin/ldconfig\n}
 			));
 		}
 		if ($OS eq "Darwin") {
@@ -462,11 +432,6 @@
 		###################################################
 		# lets do the clean:: targets at this step.
 		###################################################
-		push(@postamble, join(
-			  "\n",
-			  qq{clean ::},
-			  qq{\tcd JS/$jsdir && \$(MAKE) -f Makefile.ref BUILD_OPT=$BUILD_OPT clean\n},
-			 ));
 		if ($OS ne "Darwin") {
 			push(@postamble, join(
 			  "\n",
@@ -479,11 +444,6 @@
 		###################################################
 		# lets do the distclean:: targets at this step.
 		###################################################
-		push(@postamble, join(
-			  "\n",
-			  qq{distclean :: realclean},
-			  qq{\tcd JS/$jsdir && \$(MAKE) -f Makefile.ref BUILD_OPT=$BUILD_OPT clobber},
-			 ));
 		if ($OS ne "Darwin") {
 			push(@postamble, join(
 			  "\n",
@@ -504,8 +464,6 @@
 		if ($OS eq "Linux") {
 			push(@postamble, join(
 				"\n",
-				#qq{\tcp blib/arch/auto/VRML/VRMLFunc/VRMLFunc.so $LIB_DIR/libFreeWRLFunc.so\n},
-				#qq{\t/sbin/ldconfig\n}
 				qq{\t\tcp -f blib/arch/auto/VRML/VRMLFunc/VRMLFunc.so blib/arch/auto/VRML/VRMLFunc/libFreeWRLFunc.so\n},
 			));
 		}
