diff -urN freewrl-0.34.orig/vrml.conf freewrl-0.34/vrml.conf
--- freewrl-0.34.orig/vrml.conf	Thu Sep 19 21:47:02 2002
+++ freewrl-0.34/vrml.conf	Sun Oct 13 15:20:53 2002
@@ -16,7 +16,7 @@
     JS_INC => '-Ijs/Linux_All_DBG.OBJ -Ijs/',
 
 # Saxon location - required for x3d running.
-    SAXON_LOC => '/usr/local/src/saxon/saxon.jar',
+    SAXON_LOC => '/usr/share/java/saxon.jar',
 
 # Netscape installation
 # 
@@ -27,9 +27,9 @@
 # will be installed to.
 # NETSCAPE_PLUGINS is used to determine where the plugin will
 # be installed if this is your browser.
-    NETSCAPE_INST => '/usr/lib/netscape',
-    NETSCAPE_CLASSES => '/usr/lib/netscape/java/classes',
-    NETSCAPE_PLUGINS => '/usr/lib/netscape/plugins',
+    NETSCAPE_INST => '',
+    NETSCAPE_CLASSES => '/usr/lib/netscape/java/classes',
+    NETSCAPE_PLUGINS => '/usr/lib/netscape/plugins',
 
 # Mozilla installation
 # 
@@ -39,15 +39,15 @@
 #   MOZILLA_INST => '/usr/lib/mozilla-1.0.0',
 #   MOZILLA_PLUGINS => '/usr/lib/mozilla-1.0.0/plugins',
 #   MOZILLA_INC => '/usr/include/mozilla-1.0.0',
-    MOZILLA_INST => '',
-    MOZILLA_PLUGINS => '',
-    MOZILLA_INC => '',
+    MOZILLA_INST => '/usr/lib/mozilla',
+    MOZILLA_PLUGINS => '/usr/lib/mozilla/plugins',
+    MOZILLA_INC => '/usr/include/mozilla',
 
 ############################################################################
 # Java installation (Sun's Java 2 SDK Standard Edition 1.3.1)
 # If Java is not installed, leave JAVA_INST empty e.g.
 # JAVA_INST => ''
-    JAVA_INST => '/usr/java/j2sdk1.4.0',
-    JAVA_INC => '-I/usr/java/j2sdk1.4.0/include -I/usr/java/j2sdk1.4.0/include/linux'
+    JAVA_INST => '/usr/lib/java',
+    JAVA_INC => '-I/usr/lib/java/include -I/usr/lib/java/include/linux'
 
 );
