diff -uNr freewrl-1.17.4.orig/CFrontEnd/Makefile freewrl-1.17.4/CFrontEnd/Makefile
--- freewrl-1.17.4.orig/CFrontEnd/Makefile	2006-05-10 16:50:23.000000000 +0200
+++ freewrl-1.17.4/CFrontEnd/Makefile	2006-06-07 10:49:40.000000000 +0200
@@ -8,6 +8,7 @@
 
 install:  freewrl
 	strip freewrl
+	chmod 666 $(BINDIR)/freewrl
 	cp freewrl $(BINDIR)
 
 fwopts:
--- FreeWRL-1.12/SoundEngine/Makefile.orig	2005-04-25 15:47:19.000000000 +0200
+++ FreeWRL-1.12/SoundEngine/Makefile	2005-04-29 15:05:18.046431376 +0200
@@ -1,17 +1,16 @@
 #INCLUDE = -I/usr/X11R6/include/
 #LIBS  = -L. -L/usr/X11R6/lib -L/usr/local/lib
 
-COMPILERFLAGS = -Wall
+COMPILERFLAGS = -Wall $(OPTIMIZE)
 CC = gcc
 CFLAGS = $(COMPILERFLAGS) $(INCLUDE)
 LIBRARIES = -lm
 SVROBJ = main.o wavmaker.o mixer.o
 
-all: main
+all: FreeWRL_SoundServer
 
-main: $(SVROBJ)
+FreeWRL_SoundServer: $(SVROBJ)
 	$(CC) $(SVROBJ) $(CFLAGS) -o FreeWRL_SoundServer $(LIBS) $(LIBRARIES)
-	strip FreeWRL_SoundServer
 
 mixer: mixer.o
 	$(CC) $(CFLAGS) -o $mixer $(LIBS) $< $(LIBRARIES)
@@ -22,6 +21,6 @@
 clean:
 	rm -f *.o FreeWRL_SoundServer
 
-install:  main
+install: FreeWRL_SoundServer
 	cp FreeWRL_SoundServer $(BINDIR)
 
--- FreeWRL-1.12/FreeWRL_Message/Makefile.orig	2005-04-25 15:47:19.000000000 +0200
+++ FreeWRL-1.12/FreeWRL_Message/Makefile	2005-04-29 15:26:42.513162752 +0200
@@ -1,25 +1,21 @@
 #INCLUDE = -I/usr/X11R6/include/
 #LIBS  = -L. -L/usr/X11R6/lib -L/usr/local/lib
 
-COMPILERFLAGS = -Wall
+COMPILERFLAGS = -Wall $(OPTIMIZE)
 CC = gcc
 CFLAGS = $(COMPILERFLAGS) $(INCLUDE)
 # use this for better borders on window LIBRARIES = -lXaw3d
 LIBRARIES = -lXaw
 SVROBJ = FreeWRL_Message.o
 
-all: main
+all: FreeWRL_Message
 
-main: $(SVROBJ)
+FreeWRL_Message: $(SVROBJ)
 	$(CC) $(SVROBJ) $(CFLAGS) -o FreeWRL_Message $(LIBS) $(LIBRARIES)
-	strip FreeWRL_Message
-
-FreeWRL_Message: FreeWRL_Message.o
-	$(CC) $(CFLAGS) -o $FreeWRL_Message $(LIBS) $< $(LIBRARIES)
 
 clean:
 	rm -f *.o FreeWRL_Message
 
-install:  main
-	cp FreeWRL_Message $(BINDIR)
+install: FreeWRL_Message
+	install FreeWRL_Message $(BINDIR)
