--- freewrl-1.19.8/Makefile.PL.orig	2008-01-12 19:26:25.187644000 +0100
+++ freewrl-1.19.8/Makefile.PL	2008-01-12 19:49:11.737519783 +0100
@@ -253,14 +253,7 @@
 	print MK "JAVAC = javac\n";
 	print MK "INC = $MYINC\n";
 	print MK "DEFINE = $noStackProtector $MYDEFINE\n";
-	#print MK "FWFLAGS = -c -Wall -O2 -g -pipe -m32 -march=i386 -mtune=pentium4\n";
-
-	# different compiler flags for different architectures
-	if ($ARCH eq "x86_64") {
-		print MK "FWFLAGS = -c -O2 -g -pipe -m32 -march=i386 -mtune=pentium4\n";
-	} else {
-		print MK "FWFLAGS = -c -O2 -g -pipe -fno-strict-aliasing -fPIC\n";
-	}
+	print MK "FWFLAGS = -c -Wall \$(OPTIMIZE) -pipe -fno-strict-aliasing -fPIC\n";
 	print MK "MYLIBS = $MYLIBS\n";
 	print MK "MYFELIBS = $MYFELIBS\n";
 	print MK "MYSELIBS = $MYSELIBS\n";
@@ -386,10 +379,13 @@
 	#LIBFREEWRL
 	print MK "\$(LIBFWSRCS):\n";
 	print MK "\t\$(CC) -c \$*.c\n\n";
-	print MK "libFreeWRL:      \$(LIBFWOBJS)\n";
 	if ($OS eq "Darwin") { 
+	print MK "libFreeWRL:      libFreeWRLFunc.dylib\n\n";
+	print MK "libFreeWRLFunc.dylib:      \$(LIBFWOBJS)\n";
 	print MK "\t\$(CC) -dynamiclib \$(MYFELIBS) \$(MYLDFLAGS) -o libFreeWRLFunc.dylib  \$(MYLIBS) \$(LIBFWOBJS)\n\n";
 	} else {
+	print MK "libFreeWRL:      libFreeWRLFunc.so\n\n";
+	print MK "libFreeWRLFunc.so:      \$(LIBFWOBJS)\n";
 	print MK "\t\$(CC) -shared -o libFreeWRLFunc.so \$(LIBFWOBJS)\n\n";
 	}
 }
@@ -402,7 +398,8 @@
 	print MK "\n# make the Unix front end\n";
 	print MK "\$(FESRCS):\n";
 	print MK "\t\$(CC) -c \$*.c\n\n";
-	print MK "CFrontEnd:      \$(FEOBJS)\n";
+	print MK "CFrontEnd:      CFrontEnd/freewrl\n\n";
+	print MK "CFrontEnd/freewrl:      \$(FEOBJS)\n";
 	print MK "\t\$(CC) \$(FEOBJS) \$(MYFELIBS) -o CFrontEnd/freewrl\n\n";
 	}
 }
@@ -432,7 +429,8 @@
 	print MK "\n# make the Unix Console Message \n";
 	print MK "\$(FWMSRCS):\n";
 	print MK "\t\$(CC) -c \$*.c\n\n";
-	print MK "FreeWRL_Message:      \$(FWMOBJS)\n";
+	print MK "FreeWRL_Message:      FreeWRL_Msg/FreeWRL_Message\n\n";
+	print MK "FreeWRL_Msg/FreeWRL_Message:      \$(FWMOBJS)\n";
 	print MK "\t\$(CC) FreeWRL_Msg/FreeWRL_Message.o  -o FreeWRL_Msg/FreeWRL_Message \$(MYLIBS) \$(MYMESLIBS) \n\n";
 	}
 }
