diff -uNr freewrl-1.17.4.orig/vrml.conf freewrl-1.17.4/vrml.conf
--- freewrl-1.17.4.orig/vrml.conf	2006-06-05 09:00:13.000000000 +0200
+++ freewrl-1.17.4/vrml.conf	2006-06-05 09:07:27.000000000 +0200
@@ -95,7 +95,7 @@
 
 ############################################################################
 # Java dest. if no Java, comment out the JAVA line above, and the JAVA_INST line below.
-    JAVA_INST => '/usr/java/j2sdk1.4.0',
+    JAVA_INST => '/usr/lib/java',
 
 ############################################################################
 # LDFLAGS.  Set linker flags here.
